src/AstAnalyses.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020, The Souffle Developers. All rights reserved.
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file AstAnalyses.h
+ *
+ * Define of AST analyses classes
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "AstAnalysis.h"
+#include "AstProgram.h"
+#include "AstRelation.h"
+#include "GraphUtils.h"
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <set>
+#include <stack>
+#include <utility>
+#include <vector>
+
+#include "AstTranslationUnit.h"
+
+namespace souffle {
+
+class AstClause;
+
+/**
+ * Determine the auxiliary arity for relations
+ */
+class AuxiliaryArity : public AstAnalysis {
+public:
+    static constexpr const char* name = "auxiliary-arity";
+
+    void run(const AstTranslationUnit& translationUnit) override {
+        program = translationUnit.getProgram();
+    }
+
+    /**
+     * Returns the number of auxiliary parameters of an atom
+     * @param atom the atom (const AstAtom*)
+     * @return number of auxiliary attributes
+     */
+    const size_t getArity(const AstAtom* atom) const {
```

Comment:
Perhaps rename the methods as `getAuxiliaryArity` to better reflect their purpose, and be less confusing in code.

Reply:
I think this becomes a little bit too long since it is now renamed to auxArityAnalysis->getArity(). I feel from that context it is clear that we are dealing with auxiliary arities.